src/main/protobuf/bazel_flags.proto

```diff
@@ -46,6 +53,20 @@ message FlagInfo {
   // but if true a value must be present for all instantiations of the flag,
   // e.g. --jobs=100.
   optional bool requires_value = 10;
+  // The old, deprecated name for this option, without leading dashes.
+  optional string old_name = 11;
+  // The deprecation warning for this option, if one is present.
+  optional string deprecation_warning = 12;
+  // Additional options which are added if this option is present.
+  repeated string option_expansions = 13;
+
+  // The type of the setting
+  oneof type {
+    VoidType void = 100;
+    BooleanType boolean = 101;
+    TriStateType tri_state = 102;
+    EnumType enum = 103;
+  };
```

Comment:
should we model this as a `oneof` and essentially mirror the complete `Converter` hierarchy from Java also in Protobuf?

Or should we rather use something like

```
/// The class name of the converter as returned by `converter.getClass().getName()`
string converter_class_name = 14;
/// For enum values: the list of enum values
repeated string values = 15;
/// For range-converters: the minimum value
repeated string range_min = 16;
/// For range-converters: the maximum value
repeated string range_min = 17;
```